plugins/com.gwtplugins.gdt.eclipse.suite/src/com/google/gdt/eclipse/suite/wizards/WebAppTemplateProjectCreator.java

```diff
@@ -0,0 +1,435 @@
+      throws CoreException, SdkException, ClassNotFoundException, BackingStoreException, IOException {
+    this.monitor = monitor;
+
+    List<String> projectNames = projectTemplate.getProjectNames(projectBaseName);
+    IProject[] project = new IProject[projectNames.size()];
+    createdJavaProjects = new IJavaProject[projectNames.size()];
+    for(int i=0;i<projectNames.size();i++)
+    {
+      project[i] = createProject(monitor, projectNames.get(i));
+      File dir = URIUtil.toPath(project[i].getLocationURI()).toFile();
+      projectTemplate.copyProject(i, dir);
+      replacePackageNames(project[i], dir);
+
+      /*
+       * Refresh contents; if this project was generated via GWT's WebAppCreator, then these files would have been created
+       * directly on the file system Although, this refresh should have been done via the project.open() call, which is
+       * part of the createProject call above.
+       */
+      project[i].refreshLocal(IResource.DEPTH_INFINITE, monitor);
+
+      // Set all of the natures on the project
+      NatureUtils.addNatures(project[i], projectTemplate.getNatureIds(i));
+
+      // Create the java project
+      createdJavaProjects[i] = JavaCore.create(project[i]);
+
+      // Create a source folder and add it to the raw classpath
+      IResource warFolder = project[i].findMember(WebAppUtilities.DEFAULT_WAR_DIR_NAME);
+      boolean createWarFolders = (warFolder != null);
+      createFolders(project[i], createWarFolders, monitor);
+      if (createWarFolders) {
+        // Set the WAR source/output directory to "/war"
+        WebAppUtilities.setDefaultWarSettings(project[i]);
+
+        // Set the default output directory
+        WebAppUtilities.setOutputLocationToWebInfClasses(createdJavaProjects[i], monitor);
+        /**
+         * Copy files into the web-inf lib folder. This code assumes that it is running in a context that has a workspace
+         * lock.
+         */
+        Sdk gwtSdk = getGWTSdk();
+        setGwtSdk(gwtSdk);
+        if (gwtSdk != null) {
+          new GWTUpdateWebInfFolderCommand(createdJavaProjects[i], gwtSdk).execute();
+        }
+      }
+      reformatJavaFiles(dir);
+
+      // Allow other extensions to run after project creation
+      includeExtensionPartipants();
+    }
+  }
+
+  /**
+   * Replaces all occurrences of the PACKAGENAME with the given package name.
+   * @param iProject
+   * @param dir
+   * @throws IOException
+   */
+  private void replacePackageNames(IProject iProject, File dir) throws IOException {
+    Stack<File> files = new Stack<>();
+    files.add(dir);
+    while(files.isEmpty() == false)
+    {
+      File file = files.pop();
+      if(file.getName().contains("_PACKAGENAME_"))
+      {
+        String name = file.getName();
+        name = name.replace("_PACKAGENAME_", packageBaseName.replace('.', File.separatorChar));
+        File newFile = new File(file.getParentFile(), name);
+        newFile.getParentFile().mkdirs();
```

Comment:
consider using java.nio.Files.createDirectories(..) here - the Files class will throw instead of silently fail (or letting you ignore the return value, as you're doing here).

Reply:
changed to createDirectories